ci-operator/step-registry/openshift/e2e/gcp/libvirt/cert-rotation/conf/openshift-e2e-gcp-libvirt-cert-rotation-conf-commands.sh

```diff
@@ -0,0 +1,388 @@
+- mirrors:
+  - ${HOSTNAME}:5000/ocp4/openshift4
+  source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
+EOF
+
+# Create manifests and modify route domain
+openshift-install --dir="${CLUSTER_DIR}" create manifests
+# Workaround for https://github.com/openshift/installer/issues/1007
+# Add custom domain to cluster-ingress
+yq write --inplace $CLUSTER_DIR/manifests/cluster-ingress-02-config.yml spec[domain] apps.$BASE_DOMAIN
+
+# Add master memory to 12 GB
+# This is only valid for openshift 4.3 onwards
+yq write --inplace ${CLUSTER_DIR}/openshift/99_openshift-cluster-api_master-machines-0.yaml spec.providerSpec.value[domainMemory] 14336
+
+openshift-install create cluster --log-level=debug --dir="$CLUSTER_DIR" || true
+openshift-install wait-for install-complete --log-level=debug --dir="$CLUSTER_DIR"
+END
+chmod +x "${SHARED_DIR}"/create-cluster-mirrored-local-registry
+
+cat  > "${SHARED_DIR}"/time-skew-test.sh << 'EOF'
+#!/bin/bash
+
+set -euxo pipefail
+approveCSRs () {
+  pendingCSRs=$(oc get csr | grep Pending | wc -l)
+  if [ $pendingCSRs -ne 0 ]; then
+    echo "Approving pending csrs"
+    oc get csr -o name | xargs oc adm certificate approve
+  fi
+}
+
+checkNodesReady () {
+  nodesReady=0
+  retries=0
+  while [ $nodesReady -ne 5 ] && [ $retries -lt 30 ]; do
+    approveCSRs
+    sleep 10
+    getNodes=$(oc get nodes | grep Ready | grep -v NotReady | grep -v SchedulingDisabled | awk '{ print $1 }')
+    nodesList=( $getNodes )
+    nodesReady=${#nodesList[@]}
+    if [ $nodesReady -eq 5 ]; then
+      echo "All nodes Ready"
+    fi
+    (( retries++ ))
+  done
+  if [ $nodesReady -ne 5 ]; then
+    echo "Some nodes NotReady"
+    oc get nodes
+    exit 1
+  fi
+}
+
+sudo systemctl stop chronyd
+future=$(date -d "+370 days" +%Y-%m-%d)
+getIPs=$(oc get nodes -o wide | awk '{print $6}')
+ipList=( $getIPs )
+getDomains=$(sudo virsh list | awk '{ print $2 }' | tail -n +3 | head -n -1)
+domainList=( $getDomains )
+for i in {1..5}
+do
+  ip="${ipList[i]}"
+  ssh-keyscan -H $ip >> ~/.ssh/known_hosts
+  ssh -t core@$ip "sudo systemctl disable chronyd && sudo systemctl stop chronyd && sudo systemctl stop crio && sudo systemctl stop kubelet"
+done
+echo "setting date ahead in nodes"
+for i in {1..5}
+do
+  ip="${ipList[i]}"
+  ssh-keyscan -H $ip >> ~/.ssh/known_hosts
+  ssh -t core@$ip "sudo date --set $future"
+done
+echo "setting date on host"
+# now set date for host
+sudo date --set $future
+echo "starting kubelet, crio in nodes"
+for i in {1..5}
+do
+  ip="${ipList[i]}"
+  ssh-keyscan -H $ip >> ~/.ssh/known_hosts
+  ssh -t core@$ip "sudo systemctl start crio && sudo systemctl start kubelet"
+done
+echo "Waiting 4 minutes for csr creation..."
+sleep 240
+
+nodesReadyStable=0
+set +e
+while [ $nodesReadyStable -lt 6 ]; do
+  echo "Waiting on all nodes Ready and all stay Ready for 3 minutes..."
+  checkNodesReady
+  sleep 30
+  (( nodesReadyStable++ ))
+done
+
+echo "Checking that all ClusterOperators are Degraded=False"
```

Comment:
`oc wait co --all --for='condition=Available=True' && oc wait co --all --for='condition=Progressing=False' && oc wait co --all --for='condition=Degraded=False'`